### DISCUSSION TITLE
Optionally send document metadata to kosync server

### DISCUSSION THREAD

**Discussion by contrun** [koreader/koreader#8764] [Ideas]
In consideration of privacy and security, kosync does not send file name or file content to the backend sync server. This is great if we don't trust the backend server.

I have a different use case. I want collect all my personal information to gain new insights into my life (in spirit of [this](https://beepb00p.xyz/exports.html) and [this](https://simonwillison.net/2020/Nov/14/personal-data-warehouses/)). I have set up a kosync server for myself. Currently, the data sent to the server are quite limited. I hope we can optionally send the book metadata to the server. We can have a switch (default off) which let the users control whether to send the `metadata` field to the server.

```
{
  "percentage": 0.005,
  "device": "device_name",
  "device_id": "B78EA04ACC3A453DBA220D720C0BE102",
  "document": "348e34463a44ba68659fc6fe814a6778",
  "progress": 3,
  "metadata": {
    "filename": "Debugging: The 9 Indispensable Rules for Finding Even the Most Elusive Software and Hardware Problems",
    "author": "David J. AGANS"
  }
}
```

If this kind of metadata were also sent to the server, I may be able too gain insights of at which hour in a day I am most productive, what category of books I mostly enjoy in the night.

Any houghts on adding this switch or general ideas on how to build my personal data infrastructure?

**Comment by Frenzie** [comment]
The sync design is as minimal as possible, privacy by design. But if such an extra metadata field is silently ignored by the server I guess it could be okay.

Syncing statistics feels like a separate project, ideally doing some fairly clever things, more dumbly but perhaps significantly easier to implement just transferring the entirety of settings/statistics.sqlite3 and merging it…

**Comment by bayang** [comment]
I would be extremely interested if the official sync plugin could send some extra metadata such as title and authors because, a bit like the OP who wants to manage his own data, I'm currently building my own foss self hosted goodreads alternative and as a koreader user I was trying to find a way to automatically sync my reading progress with this app.

I reluctantly was starting to think that I would have to write a custom plugin, but actually the official sync plugin is the best place for this use case : it already has access to the data, it already has the network infrastructure, the UI etc...
An additional boolean asking if the user wants to send extra data (that data being ignored by regular kosync servers) would be a nice tradeoff, I think ?

**Reply by Frenzie** [reply]
It looks like I already agreed that should be fine back in February, depending a bit on what the impact on the code would look like.

**Comment by nperez0111** [comment]
This is resolved with https://github.com/koreader/koreader/pull/15306
